lib/system.nim

```diff
@@ -1871,6 +1871,21 @@ proc `$` *[Enum: enum](x: Enum): string {.magic: "EnumToStr", noSideEffect.}
   ## a ``$`` operator for a concrete enumeration is provided, this is
   ## used instead. (In other words: *Overwriting* is possible.)
 
+proc newString*[N](data: array[N, char]): string {.noSideEffect.} =
```

Comment:
Well actually, ``$`` works for ``cstring`` and what we need is perhaps ``proc toCstring[Idx](a: array[Idx, char]): cstring = cast[cstring](unsafeAddr(a[0]))``